### PR TITLE
Fixing some problems

### PR DESCRIPTION
**Pull request by GrateBrittan**
Should fix ponies going into the ground (Though I am unsure) and fixes people from being stuck in tase mode when you tase one person and attach it to someone else while the other person is still being tased. Though please correct me if I am wrong

### CODE REVIEW

File: lua/weapons/weapon_taser.lua [hunk 1]
```diff
@@ -24,7 +24,7 @@ if CLIENT then
         if !IsValid(ply) or !IsValid(ply.IsTaseredBy) then return end
 
         local originvec, originang = ply:GetBonePosition(1)
-        ply:ManipulateBonePosition(0, Vector(0, 0, -35))
+        ply:ManipulateBonePosition(0, Vector(0, 0, -math.abs(ply:GetViewOffset().z)))
```
Comment (swampservers) [inline lua/weapons/weapon_taser.lua]: For a human, this value is 64, for a pony, I think it's 42. Maybe you want `ply:IsPony() and -13 or -35`

File: lua/weapons/weapon_taser.lua [hunk 2]
```diff
@@ -74,6 +74,7 @@ function SWEP:PrimaryAttack() -- attempt attach to a player
     local trc = self.Owner:GetEyeTrace()
 
     if IsValid(trc.Entity) and trc.Entity:IsPlayer() and self.Owner:GetPos():DistToSqr(trc.Entity:GetPos()) < self.MaxTaseDist*self.MaxTaseDist then
+        UnTasePlayer()
```
Comment (chev2) [inline lua/weapons/weapon_taser.lua]: `UnTasePlayer()` won't work, use `self:UnTasePlayer()` instead

Comment (GrateBrittan) [reply lua/weapons/weapon_taser.lua]: Oops, sorry I didn't notice

**Comment by swampservers** [comment]
Thanks!

**Pull request merged**